beluga_amcl/src/amcl_node.cpp

```diff
@@ -706,31 +699,23 @@ std::unique_ptr<LaserLocalizationInterface2d> AmclNode::make_particle_filter(
       params.translation_noise_from_translation = get_parameter("alpha3").as_double();
       params.translation_noise_from_rotation = get_parameter("alpha4").as_double();
       params.strafe_noise_from_translation = get_parameter("alpha5").as_double();
-      return OmnidirectionalDrive{params};
+      return beluga::OmnidirectionalDriveModel{params};
     }
     if (name == kStationaryModelName) {
-      return Stationary{};
+      return beluga::StationaryModel{};
     }
     throw std::invalid_argument(std::string("Invalid motion model: ") + std::string(name));
   };
 
-  using LikelihoodField = beluga::mixin::descriptor<
-      ciabatta::curry<beluga::LikelihoodFieldModel, beluga_ros::OccupancyGrid>::mixin,
-      beluga::LikelihoodFieldModelParam>;
-
-  using BeamSensorModel = beluga::mixin::descriptor<
-      ciabatta::curry<beluga::BeamSensorModel, beluga_ros::OccupancyGrid>::mixin, beluga::BeamModelParam>;
-
-  using SensorDescriptor = std::variant<LikelihoodField, BeamSensorModel>;
-  auto get_sensor_descriptor = [this](std::string_view name) -> SensorDescriptor {
+  auto get_sensor_descriptor = [this, map](std::string_view name) -> SensorDescriptor {
     if (name == kLikelihoodFieldModelName) {
       auto params = beluga::LikelihoodFieldModelParam{};
       params.max_obstacle_distance = get_parameter("laser_likelihood_max_dist").as_double();
       params.max_laser_distance = get_parameter("laser_max_range").as_double();
       params.z_hit = get_parameter("z_hit").as_double();
       params.z_random = get_parameter("z_rand").as_double();
       params.sigma_hit = get_parameter("sigma_hit").as_double();
-      return LikelihoodField{params};
+      return beluga::LikelihoodFieldModel<beluga_ros::OccupancyGrid>{params, beluga_ros::OccupancyGrid{map}};
```

Comment:
@nahueespinosa nit: a deduction guideline would be nice. Same for other sensor models.

Reply:
Implicitly-generated deduction guides do the job actually.
Fixed in https://github.com/Ekumen-OS/beluga/pull/291/commits/deeaaa598ac33c92ae21c3a7d59d28d0e5a06ff5.